### ISSUE TITLE
Incompatible shapes:

### ISSUE THREAD

**Issue by tanmri** [keras-team/keras#9764]
I am working on chat bot. I am currently implementing 
Spoken Language Understanding ( slot fillings) for ATIS data set using LSTM.  I have stuck on a problem.  Suppose I have a list of integers length 30, 
and want to feed to LSTM and my out put label is a list of integers of length 15.

Then I got an error

InvalidArgumentError (see above for traceback): Incompatible shapes: [1,15,127] vs. [1,30,127]

I have used this structure 

# Define model
model = Sequential()
model.add(Embedding(n_vocab,127))
model.add(LSTM(127,return_sequences=True))
model.add(TimeDistributed(Dense(n_classes, activation='softmax')))  ### each output of dimension n_classes
model.compile('rmsprop', 'categorical_crossentropy')
